app/models/qdm/basetypes/ratio.rb

```diff
@@ -0,0 +1,52 @@
+module QDM
+  # Represents a Ratio
+  class Ratio
+    attr_accessor :numerator, :denominator
+
+    # Numerator and denominator are required
+    def initialize(numerator, denominator)
+      @numerator = numerator
+      @denominator = denominator
+    end
+
+    # Converts an object of this instance into a database friendly value.
+    def mongoize
+      { numerator: @numerator, denominator: @denominator, _type: 'QDM::Ratio' }
+    end
+
+    class << self
+      # Get the object as it was stored in the database, and instantiate
+      # this custom class from it.
+      #
+      # The array elements in demongoize are the same 5 elements used in mongoize, i.e.
+      # [ numerator, denominator ].
+      def demongoize(object)
+        return nil unless object
+        object = object.symbolize_keys
+        QDM::Ratio.new(object[:numerator], object[:denominator]) if object.is_a?(Hash)
+      end
+
+      # Takes any possible object and converts it to how it would be
+      # stored in the database.
+      def mongoize(object)
+        case object
+        when nil then nil
+        when QDM::Ratio then object.mongoize
+        when Hash
+          object = object.symbolize_keys
```

Comment:
Is it possible to just call `demongoize(object).mongoize` here?  These 2 lines seem somewhat repeated from lines 25-26

calling "demongoize" from within "mongoize" does seem a bit strange, so maybe it could be pulled out into a `from_hash` or something like that?

Reply:
All of the other basetypes behave the same way. I'd rather have ratio match the other basetypes since they have already proven themselves as functional.